sql/sql_alter_instance.cc

```diff
@@ -90,6 +90,14 @@ Rotate_innodb_master_key::execute()
     return true;
   }
 
+  // Acquire Percona's LOCK TABLES FOR BACKUP lock
```

Comment:
I've got only one concern. I don't know if it matters or not. I would need to analyze it deeper. Maybe @satya-bodapati will be able to answer.
Let's consider the case without PXB.
Session 1 executes `ALTER INSTANCE ROTATE INNODB MASTER KEY`. It acquires this lock.
A the same time Session 2 gets to the point when `backup_tables_lock` lock is requested, like ```if (thd->backup_tables_lock.abort_if_acquired()) return true;``` in  ```lock.cc::lock_schema_name()```. There are a lot of such paths.

Without the fix, these paths would simply acquire the lock (or not return true). Now the flow will change (parallelism will decrease).
On the flip side, I can imagine that the code responsible for MK rotation is resilient to paths that are currently acquiring `backup_tables_lock`. If we make these paths mutually exclusive, probably some logic of MK handling will become obsolete.
Yet another possibility is that MK rotation occurs rather rarely, so most probably all above is true, but does not matter :)

Reply:
@kamil-holubicki MK rotation already takes the upstream backup lock (LIFB). We extended the fix for percona's backup lock (LTFB).

Rotation is not a frequent operation IMHO.

Reply:
@kamil-holubicki not sure I get what you are saying here, but `abort_if_acquired` returns `ER_CANT_EXECUTE_WITH_BACKUP_LOCK` if LTFB has been acquired on the current session,
so there is no parallelism involved here.